### DISCUSSION TITLE
Zim version 0.74.0 released !

### DISCUSSION THREAD

**Discussion by jaap-karssenberg** [zim-desktop-wiki/zim-desktop-wiki#1570] [Announcements]
Hi all,

Today I released zim version 0.74.0. The full changelog is below. Key changes in this release are: support for "anchors" to link within pages, the ability to edit pages in a separate window and the possibility to undo edits in the last 10 pages that are edited (within a session) even after switching pages, and highly improved behavior when copy-pasting text with images from one page to another. Also there is a key user-interface change in that the main toolbar has been replaced by actions in the window border (using the Gtk3 "headerbar") and a dedicated "editing bar" underneath the page view. I realize not everyone will like this change, please see the new "Toolbar" plugin, which supports a "classic mode" to revert to something closer to the old interface.

Enjoy!

##  0.74.0 - Tue 14 Sep 2021
* Support link "anchors" to allow linking within page
* Add new-style headerbar as window decoration
* Add auto-hiding headerbar in full screen mode
* Add confirmation dialog when trashing pages
* Replace toolbar by a plugin, introduce "Edit Bar" in pageview
* Remove statusbar
* Make single page window editable
* Keep undostack for the last 10 pages
* Copy equations on copy-paste
* Copy image attachments on copy-paste
* Update relative links on copy-paste formatted text
* Fix copy-pasting links to another notebook
* Add Bullet Journal "back arrow" checkbox
* Add default browser to preferences
* Add option to paste short relative link names
* Add support for "INCLUDE file" in templates
* Add `<Ctrl><Shift>C` for copying the current line
* Show plugins that failed to load in the plugin list
* Add auto-expand and -collapse options to tags plugin
* Add option for horizontal scrollbar to pageindex & tags plugins
* Add "open backlink in new window" in backlinks plugin
* Add count and full path options for backlinks plugin
* Add rename and delete menu entries in attachmentbrowser
* Improve language selection for sourceview plugin
* Add border width preference to sourceview plugin
* Add dark mode preference to equation editor
* Add Toolbar plugin, including "classic" mode
* Add Command Palette plugin
* Also match URL in wiki text for local network
* Improve read performance for large text attachments
* Fix issue with unicode mismatch in link and page names
* Make indexer robust against false folder mtime
* Make windows installer robust against overwriting data
* Try to use GtkSourceView 4.x if 3.x not found
* Fix numbered lists with letters in Markdown export
* Use Pillow fallback to load WebP images
* Make appending to page undo-able
* Highlight "Toggle Editable" button when read-only
* Add extensions on PageWindow by splitting NotebookView and PageView
* Show link destination for keyboard navigation
* Exclude formatting when copying headings

**Comment by shivams** [comment]
Support for anchors! Yayy :love_you_gesture: This is the release many have been waiting for :smiley: Thank you Jaap :pray:

**Reply by quesada** [reply]
Really spectacular release!

**Comment by rasisin** [comment]
Thanks for all your work on this.

Regarding "Add dark mode preference to equation editor": 
This would be very helpful, but how is it done? The "Configure" button for the Equation Editor plugin doesn't do anything, and I don't see a Dark Mode option when inserting an equation.

**Reply by jaap-karssenberg** [reply]
The configure button should result in a dialog where you can check this option. If the plugin is enabled and the button does not respond please check the debug log and open a ticket if you see errors there.

**Reply by rasisin** [reply]
Thanks, it works and is definitely an improvement. Unfortunately it only affects equations created subsequently, while equations previously created remain dark. But I realize it would be more difficult to change those.

**Reply by introt** [reply]
I see a few ways to implement that. Changes could be made to either generate the needed one on page load (slow and wasteful), to generate both every time -- double the attachments, double the fun! -- or to invert the colors live, like accessibility tools do.

**Reply by introt** [reply]
Actually, after #841 gets implemented, it would totally make sense to generate both/the missing one and stick it into the cache. Wouldn't bloat attachments or slow down page load after subsequent theme toggles.

**Comment by thejeshgn** [comment]
Thank you so much for this release. UI looks great.  It gives a lot of space for the text editing.

1. A way to hide/disable bottom editing tool bar will be great.
2. I miss always available search text box

Lots of good features. Thank you again.

**Comment by jaap-karssenberg** [comment]
Toggle to hide the edit bar has been requested by more people, will look
into it.

The search box is replaced by a "search" button in the header bar. It may
feel different, but it still only requires one click to select search and
start typing. Or use Shift-Ctrl-F and start typing. So it looks different,
but number of clicks / keys is the same.

Regards,

Jaap

**Reply by thejeshgn** [reply]
Thank you

**Comment by greg4b** [comment]
Wow big update!

Looking forward for the windows build.

Thanks to the contributers!
